pkg/instrumentation/podmutator.go

```diff
@@ -42,14 +43,142 @@ type instPodMutator struct {
 	Recorder    record.EventRecorder
 }
 
+type instrumentationWithContainers struct {
+	Instrumentation *v1alpha1.Instrumentation
+	Containers      string
+}
+
 type languageInstrumentations struct {
-	Java        *v1alpha1.Instrumentation
-	NodeJS      *v1alpha1.Instrumentation
-	Python      *v1alpha1.Instrumentation
-	DotNet      *v1alpha1.Instrumentation
-	ApacheHttpd *v1alpha1.Instrumentation
-	Go          *v1alpha1.Instrumentation
-	Sdk         *v1alpha1.Instrumentation
+	Java        instrumentationWithContainers
+	NodeJS      instrumentationWithContainers
+	Python      instrumentationWithContainers
+	DotNet      instrumentationWithContainers
+	ApacheHttpd instrumentationWithContainers
+	Go          instrumentationWithContainers
+	Sdk         instrumentationWithContainers
+}
+
+// Check if single instrumentation is configured for Pod and return which is configured.
+func (langInsts languageInstrumentations) isSingleInstrumentationEnabled() bool {
+	count := 0
+
+	if langInsts.Java.Instrumentation != nil {
+		count++
+	}
+	if langInsts.NodeJS.Instrumentation != nil {
+		count++
+	}
+	if langInsts.Python.Instrumentation != nil {
+		count++
+	}
+	if langInsts.DotNet.Instrumentation != nil {
+		count++
+	}
+	if langInsts.ApacheHttpd.Instrumentation != nil {
+		count++
+	}
+	if langInsts.Go.Instrumentation != nil {
+		count++
+	}
+	if langInsts.Sdk.Instrumentation != nil {
+		count++
+	}
+
+	return count == 1
+}
+
+// Check if specific containers are provided for configured instrumentation.
+func (langInsts languageInstrumentations) areContainerNamesConfiguredForMultipleInstrumentations() (bool, error) {
+	instrWithoutContainers := 0
+	instrWithContainers := 0
+	var allContainers []string
+
+	// Check for instrumentations with and without containers.
+	if langInsts.Java.Instrumentation != nil && langInsts.Java.Containers == "" {
+		instrWithoutContainers++
+	} else if langInsts.Java.Instrumentation != nil && langInsts.Java.Containers != "" {
+		allContainers = append(allContainers, langInsts.Java.Containers)
+		instrWithContainers++
+	}
```

Comment:
We can use nested conditions so that `langInsts.Java.Instrumentation != nil` is only checked once.  We can probably also extract some duplicate logic from each language's check into a helper function.